lib/ansible/plugins/connection/persistent.py

```diff
@@ -52,9 +52,8 @@ def _do_it(self, action):
         stdin = os.fdopen(master, 'wb', 0)
         os.close(slave)
 
-        src = StringIO()
-        cPickle.dump(self._play_context.serialize(), src)
-        stdin.write(src.getvalue())
+        src = cPickle.dumps(self._play_context.serialize())
```

Comment:
By default pickle protocol for `PY3` is `3` which is not backward compatible. Need to pass `protocol` value to make it backward compatible.